code/game/machinery/autolathe/matterforge.dm

```diff
@@ -0,0 +1,594 @@
+
+
+#define ERR_OK 0
+#define ERR_NOTFOUND "not found"
+#define ERR_NOMATERIAL "no material"
+#define ERR_PAUSED "paused"
+
+/obj/machinery/matter_nanoforge
+	name = "Matter NanoForge"
+	desc = "It consumes items and produces compressed matter."
+	icon = 'icons/obj/machines/matterforge.dmi'
+	icon_state = "techprint"
+	density = TRUE
+	anchored = TRUE
+	layer = BELOW_OBJ_LAYER
+	use_power = NO_POWER_USE
+	var/list/stored_material = list()
+	var/obj/power_source
+	var/storage_capacity = 1000
+	var/unfolded
+	var/show_category
+	var/list/categories
+
+	var/list/lst = list()
```

Comment:
I think this variable should have a more descriptive name ... I think.

Reply:
done